utils/cryptutils/cryptutils_test.go

```diff
@@ -260,30 +388,57 @@ func TestCertificate(t *testing.T) {
 		t.Fatalf("Can't read file: %s", err)
 	}
 
-	if !bytes.Equal(initialCert, storedCert) {
+	if !bytes.Equal(certPem, storedCert) {
 		t.Fatal("Cert data mismatch")
 	}
 }
 
 func TestGetTLSConfig(t *testing.T) {
-	tlsDir, err := ioutil.TempDir(tmpDir, "tlsconfig")
+	caTemplate, err := GenCA()
 	if err != nil {
-		t.Fatalf("Can't create TLS config dir: %s", err)
+		t.Fatalf("Can't create CA template and keys: %s", err)
 	}
 
-	defer os.RemoveAll(tlsDir)
+	caCert, caPem, caPriv, err := testtools.GenerateCertifcateAndKeys(caTemplate, caTemplate, nil)
+	if err != nil {
+		t.Fatalf("Can't generate CA certificate: %v", err)
+	}
 
-	rootCAPath := path.Join(tlsDir, "root."+cryptutils.PEMExt)
+	template := x509.Certificate{
```

Comment:
Can we have some default template as constant in this package?

Reply:
Done